src/stravalib/model.py

```diff
@@ -499,20 +500,6 @@ def to_str_representation(
             LOGGER.warning(f"Unknown athlete type value: {raw_type}")
             return None
 
-    @lazy_property
-    def authenticated_athlete(self) -> Athlete:
-        """
-        Returns an `Athlete` object containing Strava account data
-        for the (authenticated) athlete.
-
-        Returns
-        -------
-        Athlete
-            The detailed information of the authenticated athlete.
-        """
-        assert self.bound_client is not None, "Bound client is not set."
-        return self.bound_client.get_athlete()
-
```

Comment:
Removing this (and related methods). It seems this construction aimed to prevent the `stats` property from being accessed on a non-authenticated athlete instance (which would raise a 401). But since we now know that the `stats` property is only available on the (detailed) `Athlete` class, we know for sure this is the authenticated user.